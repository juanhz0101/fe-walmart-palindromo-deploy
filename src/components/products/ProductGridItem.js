import React from 'react'
import PropTypes from 'prop-types'

function ProductGridItem( { id, brand, description, image, price, priceWithDiscount, discount} ) {

    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={image} alt={brand}/>
            <p>ID {id}</p>
            <p className="brand">Marca: {brand}</p>
            <p>Descripcion: {description}</p>
            <p>Precio original: {price}</p>
            {priceWithDiscount && <p>Precio con descuento: {priceWithDiscount} </p> }
            {discount && <p>Precio con descuento: {discount} </p> }
        </div>
    )
}

export default ProductGridItem

ProductGridItem.propTypes = {
    brand: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
}
