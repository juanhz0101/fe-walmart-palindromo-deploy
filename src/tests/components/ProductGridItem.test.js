import React from 'react';
import '@testing-library/jest-dom'
import { shallow} from 'enzyme'
import ProductGridItem from "../../components/products/ProductGridItem";


describe('Pruebas en <ProductGridItem />', () => {

    const
        id = "2", 
        brand = "dsaasd", 
        description = "zlrwax bñyrh", 
        image = "https://www.lider.cl/catalogo/images/babyIcon.svg", 
        price = "130173", 
        priceWithDiscount = "65086.5", 
        discount = "50"

    const wrapper = shallow( 
    <ProductGridItem 
        id={id} 
        brand={brand} 
        description={description} 
        image={image} 
        price={price} 
        priceWithDiscount={priceWithDiscount}
        discount={discount}
    />);

    test('debe de mostrar <ProductGridItem /> correctamente', () => {

        expect( wrapper ).toMatchSnapshot();
    })

    test('debe de tener un párrafo con la marca', () => {

        const p = wrapper.find('.brand');
        expect( p.text().trim() ).toBe( 'Marca: '+ brand );
    })

    test('debe de tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img');

        expect( img.prop('src') ).toBe( image );
        expect( img.prop('alt') ).toBe( brand );
    })

    test('debe de tener animate__fadeIn', () => {

        const div = wrapper.find('div');
        const className = div.prop('className')

        expect( className.includes('animate__fadeIn') ).toBe( true );
    })
    
})