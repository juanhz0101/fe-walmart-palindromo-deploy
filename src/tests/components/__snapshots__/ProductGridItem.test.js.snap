// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ProductGridItem /> debe de mostrar <ProductGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="dsaasd"
    src="https://www.lider.cl/catalogo/images/babyIcon.svg"
  />
  <p>
    ID 
    2
  </p>
  <p
    className="brand"
  >
    Marca: 
    dsaasd
  </p>
  <p>
    Descripcion: 
    zlrwax b√±yrh
  </p>
  <p>
    Precio original: 
    130173
  </p>
  <p>
    Precio con descuento: 
    65086.5
     
  </p>
  <p>
    Precio con descuento: 
    50
     
  </p>
</div>
`;
